### PR TITLE
Feat/v1.10.0

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

v1.10.0

## Release details

AEA:
- Makes config loader patchable
- Adds support for Python 3.10
- Enables fingerprinting for `Agent` components
- Adds support for specifying vendors when generating hashes
- Enables the usage of environment variables on `aea config` command

Plugins:
- Introduces benchmark CLI plugin

Docs:
- Add docs for benchmark CLI plugin

Chore:
- Pins correct versions on CI workflow
- Bumps `pywin32` version to `304`
- Separates tox environments for python{3.7, 3.8, 3.9} and Python3.10

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [x] Lint and unit tests pass locally and in CI
- [x] I have checked the fingerprint hashes are correct by running (`aea hash all --check`)
- [x] I have regenerated the latest API docs
- [x] I built the documentation and updated it with the latest changes
- [x] I have added an item in `HISTORY.md` for this release
- [x] I bumped the version number in the `aea/__version__.py` file.
- [x] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

### CODE REVIEW

**Review by Karrenbelt — APPROVED** [review]
LGTM

**Review by 8ball030 — APPROVED** [review]
LGTM

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
 # Release History - open AEA
 
+## 1.10.0 (2022-06-09)
+
+AEA:
+- Makes config loader patchable
+- Adds support for Python 3.10
```
Comment (DavidMinarsch) [inline HISTORY.md:7]: And removes for 3.6

File: HISTORY.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+- Enables fingerprinting for `Agent` components
```
Comment (DavidMinarsch) [inline HISTORY.md:8]:
```suggestion
- Enables fingerprinting for files in `agent` components
```

File: HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+
+Plugins:
+- Introduces benchmark CLI plugin
+
+Docs:
+- Add docs for benchmark CLI plugin
+
+Chore:
+- Pins correct versions on CI workflow
+- Bumps `pywin32` version to `304`
```
Comment (DavidMinarsch) [inline HISTORY.md:20]: also black

File: docs/upgrading.md [hunk 4]
```diff
@@ -7,6 +7,12 @@ Below we describe the additional manual steps required to upgrade between differ
 
 # Open AEA
 
+## `v1.9.0` to `v1.10.0`
+
+No backwards incompatible changes.
```
Comment (DavidMinarsch) [inline docs/upgrading.md:12]: Yes we do have: py3.6 no longer supported.

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**